### PR TITLE
Convert run-sonobuoy-test script to tekton task

### PR DESCRIPTION
**Pull request by fiunchinho**
Towards https://github.com/giantswarm/giantswarm/issues/15657

I'm still testing it, but this would be how it would look like if we decide to drop `run-sonobuoy-test` docker image and move its logic to tekton `Tasks`.

### CODE REVIEW

File: tekton/tasks/manage-sonobuoy-results.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      description: Namespace that sonobuoy will create to store its resources.
+      default: "sonobuoy"
+  steps:
+    - name: manage-sonobuoy-results
+      image: sonobuoy/sonobuoy:0.20.0
+      env:
+        - name: RESULTS_PATH
+          value: $(workspaces.cluster.path)/sonobuoy-results.tar
+      script: |
+        #! /bin/sh
```
Comment (MarcelMue) [inline tekton/tasks/manage-sonobuoy-results.yaml:25]: Not against this - just thinking:
Can we still have this as a `.sh` file which we then render in with `kustomize`?
I am not sure if that is nicer to do - but then we could have a normal .sh file

Comment (fiunchinho) [reply tekton/tasks/manage-sonobuoy-results.yaml:25]: I guess we could. I'm not sure we gain anything though. I'm open to change it if y'all feel it's better.

Comment (MarcelMue) [reply tekton/tasks/manage-sonobuoy-results.yaml:25]: I am fine either way. You can decide :)

**Comment by fiunchinho** [comment]
It works

image

File: tekton/tasks/azure.yaml [hunk 2]
```diff
@@ -11,6 +11,7 @@ spec:
   - name: test-deletion
     description: Whether or not test deletion of clusters.
     default: "1"
+    type: string
   steps:
     - name: run-tests
       image: quay.io/giantswarm/run-sonobuoy-tests
```
Comment (whites11) [inline tekton/tasks/azure.yaml:17]: But we're still using the same docker image, aren't we?

Comment (fiunchinho) [reply tekton/tasks/azure.yaml]: Arg, my bad. Fixing it.

**Review by whites11 — APPROVED** [review]
LVGTM

**Review by MarcelMue — APPROVED** [review]

**Pull request merged**